dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/get-utilisation-report-pending-corrections.controller/index.test.ts

```diff
@@ -0,0 +1,207 @@
+import httpMocks, { MockResponse } from 'node-mocks-http';
+import { HttpStatusCode } from 'axios';
+import { Response } from 'express';
+import { TestApiError, UtilisationReportEntityMockBuilder } from '@ukef/dtfs2-common';
+import { GetUtilisationReportPendingCorrectionsRequest, PendingCorrectionsResponseBody, getUtilisationReportPendingCorrectionsByBankId } from '.';
+import { getBankById } from '../../../../../repositories/banks-repo';
+import { aBank, aReportPeriod } from '../../../../../../test-helpers';
+import { UtilisationReportRepo } from '../../../../../repositories/utilisation-reports-repo';
+import { mapReportToPendingCorrectionsResponseBody } from './helpers';
+
+jest.mock('../../../../../repositories/banks-repo');
+jest.mock('../../../../../repositories/utilisation-reports-repo');
+jest.mock('./helpers');
+
+console.error = jest.fn();
+
+describe('get-utilisation-report-pending-corrections.controller', () => {
+  describe('getUtilisationReportPendingCorrections', () => {
+    const bankId = '12';
+
+    const aValidRequestQuery = () => ({ bankId });
+
+    let req: GetUtilisationReportPendingCorrectionsRequest;
+    let res: MockResponse<Response>;
+
+    beforeEach(() => {
+      req = httpMocks.createRequest<GetUtilisationReportPendingCorrectionsRequest>({
+        params: aValidRequestQuery(),
+      });
+      res = httpMocks.createResponse();
+    });
+
+    afterEach(() => {
+      jest.resetAllMocks();
+    });
+
+    it('should call getBankById with the bankId from the request', async () => {
+      // Act
+      await getUtilisationReportPendingCorrectionsByBankId(req, res);
+
+      // Assert
+      expect(getBankById).toHaveBeenCalledTimes(1);
+      expect(getBankById).toHaveBeenCalledWith(bankId);
+    });
+
+    describe('when the bank does not exist', () => {
+      beforeEach(() => {
+        jest.mocked(getBankById).mockResolvedValue(null);
+      });
+
+      it(`should respond with a '${HttpStatusCode.NotFound}' and an error message`, async () => {
+        // Act
+        await getUtilisationReportPendingCorrectionsByBankId(req, res);
+
+        // Assert
+        expect(res.statusCode).toEqual(HttpStatusCode.NotFound);
+        expect(res._getData()).toEqual(`Failed to get pending corrections: Failed to find bank with id '${bankId}'`);
+      });
+    });
+
+    describe('when the bank is not opted into utilisation reporting via DTFS', () => {
```

Comment:
should this be has not?

Reply:
I guess semantically the bank doesn't opt themselves in which I would expect to be the case with the wording 'the bank has'